### ISSUE TITLE
Tuple comparison false negative

### ISSUE THREAD

**Issue by 0xjc** [microsoft/pyright#3864]
Consider the following code. `complex` is used as an example of a type that doesn't support `__lt__`.

```
# pyright: strict


def f(a: complex, b: complex):
    # return a < b          # fails type check as expected
    return (a, a) < (b, b)  # passes type check
```

Calling `<` on `tuple[complex, complex]` and `tuple[complex, complex]` will never succeed so it should fail the type check.

Using latest pyright (1.1.268).

**Comment by erictraut** [comment]
This isn't a type error, so it wouldn't be appropriate for pyright to emit an error here.

I'll also point out that it's not quite correct to say that "Calling < on tuple[complex, complex] and tuple[complex, complex] will never succeed". Using your example above, the statement `f(1j, 1j)` works without generating an exception.

**Comment by 0xjc** [comment]
I see, could you explain what is and isn't considered a type error for cases like this? For example, `tuple[int, int] < tuple[str, str]` emits an error, but `tuple[int, str] < tuple[str, int]` does not.

**Comment by erictraut** [comment]
It's not a type error because the `tuple.__lt__` method is defined as follows:
```python
def __lt__(self, __x: tuple[_T_co, ...]) -> bool: ...
```

By contrast, there is no `complex.__lt__` method.
